### PR TITLE
intercept requester pays errors [SATURN-82]

### PR DESCRIPTION
**Pull request by panentheos**
Adds an interstitial modal in various places where we might see requester pays errors. This provides a helpful error message, cleanly exits the flow on dismissal, and provides a place where we can prompt to select a billing account in the future.

### CODE REVIEW

**Comment by elizzhao** [comment]
I'm not able to trigger the modal on data preview or notebook preview. I'm trying to with badm on "requester-pays-test" workspace

File: src/pages/workspaces/workspace/Data.js [hunk 1]
```diff
@@ -551,7 +552,10 @@ const DeleteObjectModal = ajaxCaller(class DeleteObjectModal extends Component {
   }
```
Comment (elizzhao) [inline src/pages/workspaces/workspace/Data.js:552]: I saw on the ticket that one of the AC is for downloading files from data, and I don't see that here. Is that a follow-on? Or captured elsewhere?

**Review by elizzhao — APPROVED** [review]
All these changes look good! 

I had one question though for one of the AC around downloading files from data, but that isn't a change to the current PR.

File: src/pages/workspaces/workspace/Data.js (hunk 1 above)
Comment (panentheos) [reply src/pages/workspaces/workspace/Data.js:552]: That will require some Martha work, so it will get addressed as part of SATURN-861

**Review by breilly2 — APPROVED** [review]
👍 Nice changes! I really like the approach. Thanks for the in-person walkthrough.

**Pull request merged**